rmw_connextdds_common/src/common/rmw_get_network_flow_endpoints.cpp

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021 Ericsson AB
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rmw_connextdds/rmw_impl.hpp"
+
+#include "rmw/error_handling.h"
+#include "rmw/get_network_flow_endpoints.h"
+#include "rmw/rmw.h"
+#include "rmw/types.h"
+
+extern "C"
+{
+rmw_ret_t
+rmw_publisher_get_network_flow_endpoints(
+  const rmw_publisher_t * publisher,
+  rcutils_allocator_t * allocator,
+  rmw_network_flow_endpoint_array_t * network_flow_endpoint_array)
+{
+  UNUSED_ARG(publisher);
+  UNUSED_ARG(allocator);
+  UNUSED_ARG(network_flow_endpoint_array);
+  RMW_CONNEXT_LOG_NOT_IMPLEMENTED
+  return RMW_RET_UNSUPPORTED;
+}
+
+rmw_ret_t
+rmw_subscription_get_network_flow_endpoints(
+  const rmw_subscription_t * subscription,
+  rcutils_allocator_t * allocator,
+  rmw_network_flow_endpoint_array_t * network_flow_endpoint_array)
+{
+  UNUSED_ARG(subscription);
+  UNUSED_ARG(allocator);
+  UNUSED_ARG(network_flow_endpoint_array);
+  RMW_CONNEXT_LOG_NOT_IMPLEMENTED
+  return RMW_RET_UNSUPPORTED;
+}
+}  // extern "C"
```

Comment:
Package `rmw_connextdds_common` provides the implementation of core functions shared by both `rmw_connextdds` and `rmw_connextddsmicro`.

In order to make the Windows linker happy, each RMW package must define all "public API symbols" in their own source files so that they may available in their own shared libs (instead of the ones generated by `rmw_connextdds_common`). `rmw_connextdds` does this in [rmw_api_impl_ndds.cpp](https://github.com/ros2/rmw_connextdds/blob/master/rmw_connextdds/src/rmw_api_impl_ndds.cpp), while you'll find the same for `rmw_connextddsmicro` in [rmw_api_impl_rtime.cpp](https://github.com/ros2/rmw_connextdds/blob/master/rmw_connextddsmicro/src/rmw_api_impl_rtime.cpp).

The functions in these files are just proxies for functions [defined](https://github.com/ros2/rmw_connextdds/blob/master/rmw_connextdds_common/include/rmw_connextdds/rmw_api_impl.hpp) in `rmw_connextdds_common`, and implemented in various files in `rmw_connextdds_common/src/common`.

I suggest renaming this file to `rmw_networkflows.cpp`, to keep it a bit more generic and possibly more "forward-compatible" in case there are other things to add in the future related to "unique network flows".

The functions should be defined in `rmw_api_impl.hpp`, and I would rename them to:
- `rmw_api_connextdds_publisher_get_network_flow_endpoints`
- `rmw_api_connextdds_subscription_get_network_flow_endpoints`

The implementation is fine as is.

Both `rmw_api_impl_ndds.cpp` and `rmw_api_impl_rtime.cpp` should be modified to include the actual public functions, delegating to the implementation in `rmw_connextdds_common`, e.g.:

```cpp
rmw_ret_t
rmw_publisher_get_network_flow_endpoints(
  const rmw_publisher_t * publisher,
  rcutils_allocator_t * allocator,
  rmw_network_flow_endpoint_array_t * network_flow_endpoint_array)
{
  return rmw_api_connextdds_publisher_get_network_flow_endpoints(publisher, allocator, network_flow_endpoint_array);
}
```